### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by amityadav0**
- fix opt-in
- Update readme, add typescript section, add project description

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -60,3 +60,16 @@ These files are called checkpoints and they save information about deployed asse
 As they are checked-in into Version Control System you can use them to track the deployed state of your assets.
 
 Read more about checkpoints [here](/docs/execution-checkpoints.md).
+
+### Typescript Projects
+
+`Typescript` project requires a compilation to `js` files. So that algob can execute `js` files.
+To develop in `typescript`, please remember these points:
+
+- Make sure to compile your `ts` project using `tsc --build .` Never forget to compile files.
```
Comment (ratik21) [inline docs/README.md:69]:
```suggestion
- Make sure to compile your `ts` project using `yarn build` (which runs `tsc --build .`) Never forget to compile files.
```

File: docs/README.md [hunk 2]
```diff
@@ -60,3 +60,16 @@ These files are called checkpoints and they save information about deployed asse
...
+- If you are actively developing, please use `tsc -w -p .` to build your files in real time.
```
Comment (ratik21) [inline docs/README.md:70]:
```suggestion
- TIP: If you are actively developing, please use `yarn build:watch`(`tsc -w -p .`) to build or compile your `.ts` files in real time.
```

File: docs/README.md [hunk 3]
```diff
@@ -60,3 +60,16 @@ These files are called checkpoints and they save information about deployed asse
...
+- Best way to build project is to run tsc in a watch mode.
```
Comment (ratik21) [inline docs/README.md:71]: Is this different from the above point?

Comment (amityadav0) [reply docs/README.md:71]: yes, it says the below method is better

Comment (ratik21) [reply docs/README.md:71]: We could just add `(this is recommended)` in the above point. Seems more intuitive to me

Comment (amityadav0) [reply docs/README.md:71]: Fair point.

File: examples/README.md [hunk 4]
```diff
@@ -44,3 +44,17 @@ copy the `/examples/algob.confg-template.js` to `/examples/algob.config-local.js
...
+- [permissioned-voting](./permissioned-voting) -  This `javascript` project demonstrates how to create a Permissioned Voting Stateful Smart Contract Application.
+Original tutorial can be found [here](https://developer.algorand.org/solutions/example-permissioned-voting-stateful-smart-contract-application/)
+- [ref-templates](./ref-templates) - The goal of this project is to present how to work with ASC in `algob` using the best practices and templates using the Algorand reference templates.This is a `javascript` project.
+- [signed-txn](./signed-txn) - This example demonstrates executing a serialized transaction loaded directly from a file. A user can create a transaction and sign it (using `goal`) and send it to someone else to execute it. This is a `javascript` project.
+- [htlc](./htlc-pyteal-ts) - Hash-Time-Lock-Contract Example in `typescript` using PyTeal.
+This is a `typescript` project therefore it requires a compilation to `js` files.
+In this project we are transpiling the files in `js` and then using them with `algob`.
```
Comment (ratik21) [inline examples/README.md:60]: In the above point we are using `compiling`, and here we are using `transpiling`. Can we use same methodology throughout? Also, not sure but it seems this line is similar to the previous one.

File: examples/README.md [hunk 5]
```diff
@@ -44,3 +44,17 @@ copy the `/examples/algob.confg-template.js` to `/examples/algob.config-local.js
...
+
+### Examples Description
+
+- [asa](./asa) - This `javascript` project shows how to create Algorand Standard Asset (ASA).
+- [crowdfunding](./crowdfunding) - This `javascript` project demonstrates how to create a Crowdfunding Stateful Smart Contract Application. It's based on a [tutorial](https://developer.algorand.org/solutions/
+- [multisig](./multisig) - This `javascript` example demonstrates authorizing transactions based on logic signature signed by multiple accounts.
+- [nft](./nft) - Non-Fungible-Token Example using stateful TEAL in `javascript`. In this example, we create a new non-fungible-token represented by a name and a ref.
+- [permissioned-voting](./permissioned-voting) -  This `javascript` project demonstrates how to create a Permissioned Voting Stateful Smart Contract Application.
+Original tutorial can be found [here](https://developer.algorand.org/solutions/example-permissioned-voting-stateful-smart-contract-application/)
+- [ref-templates](./ref-templates) - The goal of this project is to present how to work with ASC in `algob` using the best practices and templates using the Algorand reference templates.This is a `javascript` project.
```
Comment (ratik21) [inline examples/README.md:56]: I think except `htlc` all projects are `js` projects. Can we list them in separate subsections? 
(`# Javascript [projects..], # Typescript [projects..]`)

File: examples/README.md [hunk 6]
```diff
@@ -44,3 +44,17 @@ copy the `/examples/algob.confg-template.js` to `/examples/algob.config-local.js
 the following positions in the latter file:
 
 + `master-account`: must be an account with some ALGOs - it will be used for deployment and for activating / funding other accounts.
+
+### Examples Description
+
+- [asa](./asa) - This `javascript` project shows how to create Algorand Standard Asset (ASA).
+- [crowdfunding](./crowdfunding) - This `javascript` project demonstrates how to create a Crowdfunding Stateful Smart Contract Application. It's based on a [tutorial](https://developer.algorand.org/solutions/
```
Comment (ratik21) [inline examples/README.md:51]: Is the link here correct?

File: examples/README.md [hunk 7]
```diff
@@ -44,3 +44,17 @@ copy the `/examples/algob.confg-template.js` to `/examples/algob.config-local.js
...
+- [multisig](./multisig) - This `javascript` example demonstrates authorizing transactions based on logic signature signed by multiple accounts.
+- [nft](./nft) - Non-Fungible-Token Example using stateful TEAL in `javascript`. In this example, we create a new non-fungible-token represented by a name and a ref.
+- [permissioned-voting](./permissioned-voting) -  This `javascript` project demonstrates how to create a Permissioned Voting Stateful Smart Contract Application.
```
Comment (ratik21) [inline examples/README.md:54]: Please also add link to tutorial here

File: examples/README.md [hunk 8]
```diff
@@ -44,3 +44,17 @@ copy the `/examples/algob.confg-template.js` to `/examples/algob.config-local.js
 the following positions in the latter file:
 
 + `master-account`: must be an account with some ALGOs - it will be used for deployment and for activating / funding other accounts.
+
+### Examples Description
+
+- [asa](./asa) - This `javascript` project shows how to create Algorand Standard Asset (ASA).
```
Comment (ratik21) [inline examples/README.md:50]:
```suggestion
- [asa](./asa) - This `javascript` project demonstrates how to create and manage Algorand Standard Assets (ASA).
```

File: docs/README.md [hunk 9]
```diff
@@ -60,3 +60,15 @@ These files are called checkpoints and they save information about deployed asse
...
 
 Read more about checkpoints [here](/docs/execution-checkpoints.md).
+
+### Typescript Projects
+
+`Typescript` project requires a compilation to `js` files. So that algob can execute `js` files.
+To develop in `typescript`, please remember these points:
+
+- Make sure to compile your `ts` project using `yarn build` (which runs `tsc --build .`) Never forget to compile files.
+- TIP: If you are actively developing, please use `yarn build:watch`(`tsc -w -p .`) to build or compile your `.ts` files in real time.(this is recommended)
```
Comment (ratik21) [inline docs/README.md:70]:
```suggestion
- TIP: If you are actively developing, please use `yarn build:watch`(`tsc -w -p .`) to build or compile your `.ts` files in real time (this is recommended).
```

File: examples/README.md (hunk 4 above)
Comment (amityadav0) [reply examples/README.md:60]: ok, changing to transpiling, they seem different to me.

File: examples/README.md (hunk 5 above)
Comment (amityadav0) [reply examples/README.md:56]: i don't have preference. can separate as well ?

Comment (ratik21) [reply examples/README.md:56]: We are repeatedly using `this is javascript project` in these points, so i thought having separate section could be better. I don't have a strong preference, we can keep this as well.

Comment (robert-zaremba) [reply examples/README.md]: How about using a tag? I will be more readable. We can do it like this:

- [JS] [permissioned-voting](./permissioned-voting) - project that demonstrates how to create a Permissioned Voting Stateful Smart Contract Application.
- [JS] [link](link) - other project description
- [TS] [link](link) - other project description

Comment (amityadav0) [reply examples/README.md]: yes

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**